UI/window-basic-settings-a11y.cpp

```diff
@@ -209,191 +209,36 @@ void OBSBasicSettings::on_colorPreset_currentIndexChanged(int idx)
-
-	UpdateA11yColors();
-}
-
-void OBSBasicSettings::on_choose4_clicked()
-{
-	QColor color = GetColor(
-		mixerGreen,
-		QTStr("Basic.Settings.Accessibility.ColorOverrides.MixerGreen"));
-
-	if (!color.isValid())
-		return;
-
-	mixerGreen = color_to_int(color);
-
-	preset = COLOR_PRESET_CUSTOM;
-	bool block = ui->colorPreset->blockSignals(true);
-	ui->colorPreset->setCurrentIndex(ui->colorPreset->count() - 1);
-	ui->colorPreset->blockSignals(block);
-
-	A11yChanged();
-
-	UpdateA11yColors();
-}
-
-void OBSBasicSettings::on_choose5_clicked()
-{
-	QColor color = GetColor(
-		mixerYellow,
-		QTStr("Basic.Settings.Accessibility.ColorOverrides.MixerYellow"));
-
-	if (!color.isValid())
-		return;
-
-	mixerYellow = color_to_int(color);
-
-	preset = COLOR_PRESET_CUSTOM;
-	bool block = ui->colorPreset->blockSignals(true);
-	ui->colorPreset->setCurrentIndex(ui->colorPreset->count() - 1);
-	ui->colorPreset->blockSignals(block);
-
-	A11yChanged();
-
-	UpdateA11yColors();
-}
-
-void OBSBasicSettings::on_choose6_clicked()
-{
-	QColor color = GetColor(
-		mixerRed,
-		QTStr("Basic.Settings.Accessibility.ColorOverrides.MixerRed"));
-
-	if (!color.isValid())
-		return;
-
-	mixerRed = color_to_int(color);
-
-	preset = COLOR_PRESET_CUSTOM;
-	bool block = ui->colorPreset->blockSignals(true);
-	ui->colorPreset->setCurrentIndex(ui->colorPreset->count() - 1);
-	ui->colorPreset->blockSignals(block);
-
-	A11yChanged();
-
-	UpdateA11yColors();
-}
-
-void OBSBasicSettings::on_choose7_clicked()
-{
-	QColor color = GetColor(
-		mixerGreenActive,
-		QTStr("Basic.Settings.Accessibility.ColorOverrides.MixerGreenActive"));
-
-	if (!color.isValid())
-		return;
-
-	mixerGreenActive = color_to_int(color);
-
-	preset = COLOR_PRESET_CUSTOM;
-	bool block = ui->colorPreset->blockSignals(true);
-	ui->colorPreset->setCurrentIndex(ui->colorPreset->count() - 1);
-	ui->colorPreset->blockSignals(block);
-
-	A11yChanged();
-
-	UpdateA11yColors();
-}
-
-void OBSBasicSettings::on_choose8_clicked()
-{
-	QColor color = GetColor(
-		mixerYellowActive,
-		QTStr("Basic.Settings.Accessibility.ColorOverrides.MixerYellowActive"));
-
-	if (!color.isValid())
-		return;
-
-	mixerYellowActive = color_to_int(color);
-
-	preset = COLOR_PRESET_CUSTOM;
-	bool block = ui->colorPreset->blockSignals(true);
-	ui->colorPreset->setCurrentIndex(ui->colorPreset->count() - 1);
-	ui->colorPreset->blockSignals(block);
-
-	A11yChanged();
-
-	UpdateA11yColors();
-}
-
-void OBSBasicSettings::on_choose9_clicked()
-{
-	QColor color = GetColor(
-		mixerRedActive,
-		QTStr("Basic.Settings.Accessibility.ColorOverrides.MixerRedActive"));
-
-	if (!color.isValid())
-		return;
-
-	mixerRedActive = color_to_int(color);
-
-	preset = COLOR_PRESET_CUSTOM;
-	bool block = ui->colorPreset->blockSignals(true);
-	ui->colorPreset->setCurrentIndex(ui->colorPreset->count() - 1);
-	ui->colorPreset->blockSignals(block);
-
-	A11yChanged();
+#define GENERATE_ON_CHOOSE_CLICKED(color_override_var, color_override_str,                         \
```

Comment:
Let's not do this. Macros are generally hard to read, debug and maintain.
If anything, create a function and connect the buttons to it manually, maybe in the style of https://github.com/obsproject/obs-studio/blob/075f3c40207fd29fa80ffd9eb8fd1396866374d0/UI/window-basic-main.cpp#L448-L467